hermes-consumers/src/main/java/pl/allegro/tech/hermes/consumers/supervisor/ConsumerFactory.java

```diff
@@ -63,15 +67,18 @@ Consumer createConsumer(Subscription subscription) {
 
         Semaphore inflightSemaphore = new Semaphore(configFactory.getIntProperty(CONSUMER_INFLIGHT_SIZE));
 
+        Topic topic = topicRepository.getTopicDetails(subscription.getTopicName());
+
         return new Consumer(
-            new SplitMessagesReceiver(messageReceiverFactory.createMessageReceiver(subscription), messageSplitter),
+            messageReceiverFactory.createMessageReceiver(topic.getContentType(), subscription),
```

Comment:
maybe we should pass just `topic` instead of `topic.getContentType()`, which means every layer has to know that topic content-type may vary